node/pkg/watchers/sui/watcher.go

```diff
@@ -433,11 +441,21 @@ func (e *Watcher) Run(ctx context.Context) error {
 				}
 				resp.Body.Close()
 
-				logger.Info("receive", zap.String("body", string(body)))
+				logger.Debug("receive", zap.String("body", string(body)))
 
-				if strings.Contains(string(body), "error") {
-					logger.Error("Failed to get events for re-observation request", zap.String("Result", string(body)))
-					continue
+				// Do we have an error?
+				var err_res SuiTxnQueryError
+				err = json.Unmarshal(body, &err_res)
+				if err != nil {
+					logger.Error("Failed to unmarshal event error message", zap.String("Result", string(body)))
+					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+					return err
+				}
+
+				if err_res.Error.Message != nil {
+					logger.Error("Failed to get events for re-observation request, detected error", zap.String("Result", string(body)))
+					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+					return errors.New(*err_res.Error.Message)
```

Comment:
I don't think we want to kill the watcher here.

Reply:
Fixed